packages/react-components/react-menu/src/components/Menu/Menu.types.ts

```diff
@@ -162,6 +162,8 @@ export type MenuOpenChangeData = {
   bubble?: boolean;
   /**
    * Indicates whether the change of state was a keyboard interaction
+   * @deprecated
+   * This should not be used, since `Escape`, `Space` and click should be interpreted as the same thing as a click
```

Comment:
```suggestion
   * @deprecated
   * This should not be used, since `Enter`, `Space` and click should be interpreted as the same thing as a click
```

Escape is quite rightly a keyboard event. However you can state that it's better to get the relevant event data directly from the event